packages/cojson/src/sync.ts

```diff
@@ -514,39 +478,25 @@ export class SyncManager {
             }
         }
 
-        if (entry.state === "loading") {
+        if (entry.state.type === "unknown") {
             const availableOnPeer = peer.optimisticKnownStates[msg.id]?.header;
-            const firstPeerStateEntry = entry.firstPeerState[peer.id];
-            if (firstPeerStateEntry?.type === "waiting") {
-                firstPeerStateEntry.resolve();
-            }
-            entry.firstPeerState[peer.id] = availableOnPeer
-                ? { type: "available" }
-                : { type: "unavailable" };
-            // console.log(
-            //     "Marking",
-            //     msg.id,
-            //     "as",
-            //     entry.firstPeerState[peer.id]?.type,
-            //     "from",
-            //     peer.id
-            // );
-            if (
-                Object.values(entry.firstPeerState).every(
-                    (s) => s.type === "unavailable",
-                )
-            ) {
-                entry.resolve("unavailable");
+
+            if (!availableOnPeer) {
+                entry.dispatch({
+                    type: "not-found",
+                    peerId: peer.id,
+                });
             }
```

Comment:
I'm not updating the peer state to "available" because:
- notifying the listeners by resolving the peer state promise breaks the sync
- the coValue is thruly available in the peer if we have got it's headers